File: docs-examples/hello-world-java/src/test/java/io/micronaut/graphql/docs/GraphQLDocsTest.java
package io.micronaut.graphql.docs;

import io.micronaut.http.HttpRequest;
import io.micronaut.http.MutableHttpRequest;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

@MicronautTest
class GraphQLDocsTest {

    @Inject
    @Client("/")
    HttpClient client;

    @Test
    void graphQlWorksAsExpected() {
        MutableHttpRequest<Object> get = HttpRequest.GET("graphql");
        get.getParameters().add("query", "query { hello }");

        Map retrieve = client.toBlocking().retrieve(get, Map.class);

        assertEquals("Hello World!", ((Map)retrieve.get("data")).get("hello"));
    }
}

File: docs-examples/hello-world-kotlin/src/test/kotlin/io/micronaut/graphql/docs/GraphQLDocsTest.kt
package io.micronaut.graphql.docs

import io.micronaut.http.HttpRequest
import io.micronaut.http.client.HttpClient
import io.micronaut.http.client.annotation.Client
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

@MicronautTest
class GraphQLDocsTest {

    @Inject
    @field:Client("/")
    lateinit var client: HttpClient

    @Test
    fun graphQlWorksAsExpected() {
        val get = HttpRequest.GET<Any>("graphql").apply {
            parameters.add("query", "query { hello }")
        }

        val retrieve = client.toBlocking().retrieve(get, Map::class.java)

        Assertions.assertEquals(
            "Hello World!",
            (retrieve["data"] as Map<*, *>)["hello"]
        )
    }
}
